method/link.js

```diff
@@ -84,6 +84,50 @@ module.exports = function Link (ssb) {
         }),
         pull.collect(cb)
       )
+    },
+    findSubgroupByGroupId (groupId, cb) {
+      if (!isGroup(groupId)) return cb(new Error(`findSubgroupByGroupId expected a groupId, got ${groupId} instead.`))
+
+      const query = [{
+        $filter: {
+          value: {
+            content: {
+              type: 'link/group-subgroup',
+              parent: groupId,
+              tangles: {
+                link: { root: null, previous: null }
+              }
+            }
+          }
+        }
+      }]
+
+      pull(
+        ssb.query.read({ query }),
+        pull.filter(groupSubgroupLink.spec.isRoot),
+        // commented this out as its making it fail!
+        // pull.filter(link => {
```

Comment:
For some reason, this filter way filtering out valid links and causing the test to fail. Not sure why though!!

When ever the return part was false, it filtered the link out

Reply:
very strange. this look valid to me, except you might need to be careful about assuming that recps are present.

OH, you should be comparing `recps[0]` with `parent`. This is assuming that you're making a link which is visible to the people in the Parent group.... 
This goes against our previous pattern where I think we generally encrypted links to the child context....

Reply:
Not sure what you mean when you say:
`This goes against our previous pattern where I think we generally encrypted links to the child context....`